src/ccall.cpp

```diff
@@ -1481,6 +1481,22 @@ static jl_cgval_t emit_ccall(jl_codectx_t &ctx, jl_value_t **args, size_t nargs)
         tbaa_decorate(tbaa_const, tid);
         return mark_or_box_ccall_result(ctx, tid, retboxed, rt, unionall, static_rt);
     }
+    else if (is_libjulia_func(jl_gc_disable_finalizers_internal)
+#ifdef NDEBUG
+             || is_libjulia_func(jl_gc_enable_finalizers_internal)
+#endif
+             ) {
+        JL_GC_POP();
+        Value *ptls_i32 = emit_bitcast(ctx, ctx.ptlsStates, T_pint32);
+        const int finh_offset = offsetof(jl_tls_states_t, finalizers_inhibited);
+        Value *pfinh = ctx.builder.CreateInBoundsGEP(ptls_i32, ConstantInt::get(T_size, finh_offset / 4));
+        LoadInst *finh = ctx.builder.CreateAlignedLoad(pfinh, Align(sizeof(int32_t)));
+        if (is_libjulia_func(jl_gc_disable_finalizers_internal))
+            ctx.builder.CreateStore(ctx.builder.CreateAdd(finh, ConstantInt::get(T_int32, 1)), pfinh);
+        else
+            ctx.builder.CreateStore(ctx.builder.CreateSub(finh, ConstantInt::get(T_int32, 1)), pfinh);
```

Comment:
Emit `llvm.usub.sat.i32` here? Or 
```suggestion
            ctx.builder.CreateStore(ctx.builder.CreateSelect(ctx.builder.CreateICmpEQ(finh, ConstantInt::get(T_int32, 0)),
                ConstantInt::get(T_int32, 0),
                ctx.builder.CreateSub(finh, ConstantInt::get(T_int32, 1)), pfinh);
```